app/viewModels/houseConsignment/index/items/additionalReference/AddAnotherAdditionalReferenceViewModel.scala

```diff
@@ -69,15 +70,18 @@ object AddAnotherAdditionalReferenceViewModel {
         .flatMap {
           case (_, i) =>
             val additionalReferenceIndex = Index(i)
-            val number                   = userAnswers.get(AdditionalReferenceNumberPage(additionalReferenceIndex)).getOrElse("")
-            val numberString             = if (number.nonEmpty) s"- $number" else ""
-            userAnswers.get(AdditionalReferenceTypePage(additionalReferenceIndex)).map {
+            val numberString = userAnswers.get(AdditionalReferenceNumberPage(houseConsignmentIndex, itemIndex, additionalReferenceIndex)) match {
```

Comment:
Very minor but val could be made lazy as only accessed when AdditionalReferenceTypePage populated at the relevant indices